### PR TITLE
Implemented the share feature (#11)

### PR DESCRIPTION
**Pull request by ShivamMenda**
Also changed the Colour Picker version because it was giving a warning of Decraparation.

### CODE REVIEW

**Comment by ShivamMenda** [comment]
@thisisamank I have implemented the share feature. Please do a review and let me know if there are any errors. Waiting for your feedback and validation. Thank you.

File: lib/view_models/select_image/image_view_model.dart [hunk 1]
```diff
@@ -84,6 +87,22 @@ class SelectImageViewModel extends ChangeNotifier {
...
+    takePicture();
+  }
+
+  Future<void> takePicture() async {
+    RenderRepaintBoundary boundary = _screenshotKey.currentContext!
+        .findRenderObject() as RenderRepaintBoundary;
+    ui.Image image = await boundary.toImage();
+    final directory = (await getApplicationDocumentsDirectory()).path;
+    ByteData? byteData = await image.toByteData(format: ui.ImageByteFormat.png);
+    Uint8List pngBytes = byteData!.buffer.asUint8List();
```
Comment (thisisamank) [inline lib/view_models/select_image/image_view_model.dart]: This code is repeated, may be refactor the previous `_convertImageToWidget()` to `_convertImageToBytes()` and return `Uint8List` from it. Use it in the `saveImage` method. Then use the same method for `takePicture()`  method. That would be a cleaner way to do this I guess.

File: lib/view_models/select_image/image_view_model.dart [hunk 2]
```diff
@@ -10,8 +10,11 @@ import 'package:flutter/rendering.dart';
 import 'package:flutter/services.dart';
 import 'package:image_gallery_saver/image_gallery_saver.dart';
 import 'package:permission_handler/permission_handler.dart';
+import 'package:share/share.dart';
+import 'package:path_provider/path_provider.dart';
 
 class SelectImageViewModel extends ChangeNotifier {
+  final GlobalKey genKey = GlobalKey();
```
Comment (thisisamank) [inline lib/view_models/select_image/image_view_model.dart]: What is the use of this?

**Review by thisisamank — CHANGES_REQUESTED** [review]

File: lib/view_models/select_image/image_view_model.dart (hunk 1 above)
Comment (ShivamMenda) [reply lib/view_models/select_image/image_view_model.dart]: Okay, will try using refactoring.

File: lib/view_models/select_image/image_view_model.dart (hunk 2 above)
Comment (ShivamMenda) [reply lib/view_models/select_image/image_view_model.dart]: Oh yeah, I was trying some approach from a tutorial, It didn't work so i'll remove this.

**Comment by ShivamMenda** [comment]
@thisisamank I have refactored the code. Please do check it out. Thanks.

**Review by thisisamank — APPROVED** [review]
Thanks a lot for your contribution! There were a few minor tweaks required which I did. There are a few bugs on master branch so once I resolve them I will merge your PR.

**Comment by ShivamMenda** [comment]
Okay @thisisamank Sure. Thank you.

**Comment by thisisamank** [comment]
Hey @ShivamMenda There are a few issues at my side. My master branch was bit messed up so I had to move to last stable version.
I'm sure you must be having the code. Can you make a fresh PR with your changes? 
Really sorry for the trouble.

**Comment by ShivamMenda** [comment]
Hey @thisisamank Let me check about it and get back to you. I'll make a fresh PR with my changes.

**Comment by thisisamank** [comment]
Hey @ShivamMenda were you able to check it? You can also copy relevant code (skip the code related to dark mode commits) from here and make a fresh commit to the original repo. Only reason I didn't do it because the commit will have me as author and you have worked on this issue so that's why ;)

**Comment by ShivamMenda** [comment]
Hey @thisisamank will this affect my hacktoberfest PR's as I open another pull request? I was also thinking if copy pasting from here only

**Comment by ShivamMenda** [comment]
@thisisamank if this doesn't affect my PR's and contributions towards hacktoberfest I'll open a fresh PR.

**Comment by ShivamMenda** [comment]
@thisisamank Actually you can only make the commits if you want to. Anyway my PR is merged so it will count as a contribution.

**Comment by thisisamank** [comment]
No no, it's completely fine. 2 PRs will be counted.

**Comment by ShivamMenda** [comment]
@thisisamank okay will open a fresh PR tommorow with the changes.

**Comment by ShivamMenda** [comment]
@thisisamank I'm so sorry for the delay but there has been something wrong with my laptop and it has gone for service and it will take a while before it comes back. I think it's best you only do the commits from this PR. I'm not sure when the laptop would come back. So it's better that you only commit off the changes. Sorry again.

**Pull request merged**